vm/systemSmartContracts/auction.go

```diff
@@ -1386,6 +1388,50 @@ func (s *stakingAuctionSC) EpochConfirmed(epoch uint32) {
 	log.Debug("stakingAuctionSC: auction", "enabled", s.flagAuction.IsSet())
 }
 
+func (s *stakingAuctionSC) getBlsKeysStatus(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !bytes.Equal(args.CallerAddr, s.auctionSCAddress) {
+		s.eei.AddReturnMessage("this is only a view function")
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) != 1 {
+		s.eei.AddReturnMessage("number of arguments must be equal to 1")
+		return vmcommon.UserError
+	}
+
+	registrationData, err := s.getOrCreateRegistrationData(args.Arguments[0])
+	if err != nil {
+		s.eei.AddReturnMessage("cannot get or create registration data: error " + err.Error())
+		return vmcommon.UserError
+	}
+
+	if len(registrationData.BlsPubKeys) == 0 {
+		s.eei.AddReturnMessage("no bls keys")
+		return vmcommon.Ok
+	}
+
+	for _, blsKey := range registrationData.BlsPubKeys {
+		vmOutput, err := s.executeOnStakingSC([]byte("getBLSKeyStatus@" + hex.EncodeToString(blsKey)))
+		if err != nil {
+			s.eei.AddReturnMessage("cannot get bls key status: bls key - " + hex.EncodeToString(blsKey) + " error - " + err.Error())
+			continue
+		}
+
+		if vmOutput.ReturnCode != vmcommon.Ok {
+			continue
```

Comment:
Do we not need also a s.eei.AddReturnMessage here?

Reply:
should not happen.